.github/workflows/build.yml

```diff
@@ -33,25 +38,12 @@ jobs:
         uses: arma-actions/hemtt@master
         with:
           command: build --release --ci
-      # - name: Upload Artifact # Disabled to avoid warnings and it is useless right now.
-      #   uses: actions/upload-artifact@konradpabjan/v2-is-almost-here
-      #   with:
-      #     name: zen-${{ github.sha }}
-      #     path: releases/@zen-*.zip
-  notify:
-    needs: [validate, build]
-    if: always()
-    runs-on: ubuntu-latest
-    steps:
-      - name: Notify Discord
+      - name: Send Discord notification
+        uses: 8398a7/action-slack@v2
+        if: failure()
```

Comment:
These jobs happen concurrently, so if both fail then you get double notifications.

Reply:
But they are different jobs, if Validation fails then Build may succeed.

To be frank, there are really two options:
1. This approach for notifications if one job fails (if you write really bad code that causes both jobs to fail then, yes, you'll get double notifications)
2. Continue to use Rezent/Mechilles which is an external application that has to be kept up to date too.